src/vehicle.h

```diff
@@ -734,24 +741,32 @@ class vehicle
         vehicle_part_range get_parts() const;
         /**
          * Yields a range of parts of this vehicle that each have the given feature
-         * and are not broken. Removed parts are also excluded. The enabled status
-         * of the part is ignored.
+         * and are available: not broken, removed, or part of a carried vehicle.
+         * The enabled status * of the part is ignored.
          */
         /**@{*/
-        vehicle_part_with_feature_range<std::string> get_parts( std::string feature ) const;
-        vehicle_part_with_feature_range<vpart_bitflags> get_parts( vpart_bitflags f ) const;
+        vehicle_part_with_feature_range<std::string> get_avail_parts( std::string feature ) const;
+        vehicle_part_with_feature_range<vpart_bitflags> get_avail_parts( vpart_bitflags f ) const;
         /**@}*/
         /**
-         * Like @ref get_parts, but also yields broken parts.
+         * Like @ref get_avail_parts, but also yields parts on carried vehicles.
```

Comment:
These functions do not call each other (`get_parts_including_carried` does not call `get_avail_parts`), yet they reference each other. This can easily go wrong: if someone changes `get_avail_parts` (but not the other function), they will probably net check what other documentation referees to it. So the current comment ("like `get_avail_parts`") may not be correct anymore.

Beside that it gets a bit difficult to understand: `get_avail_parts` lists a bunch of attributes that will exclude the part, and the function comments below list a bunch of attributes that suddenly include that part again.

Ideally each function would simply list what attributes are required.

Reply:
I was copying the style of the docs but it was bad as you pointed out.

Fixed.